### PR TITLE
SSO Redirect

### LINKED ISSUE #91: Implement SSO for discourse
# Feature
## Why is this feature being added?

In order to provide a seamless UX between oc.org and discourse we need to implement SSO for discourse.

## What should your feature do?
We should send all info needed for SSO discourse into the JWT.

These links should help:
https://meta.discourse.org/t/official-single-sign-on-for-discourse/13045
https://stackoverflow.com/questions/25478510/incorporating-discourse-sso-with-existing-rails-site-with-devise

### PR DESCRIPTION
**Pull request by rickr**
# Description of changes

Adds logic for discourse SSO logins

# Issue Resolved

Fixes #91

### CODE REVIEW

File: app/controllers/api/v1/sessions_controller.rb [hunk 1]
```diff
@@ -5,17 +5,43 @@ class SessionsController < Devise::SessionsController
...
       end
 
       def set_default_response_format
         request.format = 'json'
       end
 
+      private
+
+      def sso_request?
+        return true if params[:sso].present? && params[:sig].present?
```
Comment (hpjaj) [inline app/controllers/api/v1/sessions_controller.rb]: Perhaps remove the `return true if` to return either `boolean` case:

```ruby
def sso_request?
  params[:sso].present? && params[:sig].present?
end

File: app/lib/discourse/single_sign_on.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    def self.sso_secret
+      raise RuntimeError, "sso_secret not implemented on class, be sure to set it on instance"
+    end
+
+    def self.sso_url
+      raise RuntimeError, "sso_url not implemented on class, be sure to set it on instance"
+    end
+
+    def self.parse(payload, sso_secret = nil)
```
Comment (hpjaj) [inline app/lib/discourse/single_sign_on.rb]: This method has a few chunks of logic.  Can you break some of it out into smaller, single purpose `private` methods?

File: app/controllers/api/v1/sessions_controller.rb (hunk 1 above)
Comment (rickr) [inline app/controllers/api/v1/sessions_controller.rb]: 7ec989a

File: app/lib/discourse/single_sign_on.rb (hunk 2 above)
Comment (rickr) [inline app/lib/discourse/single_sign_on.rb]: I don't love this code either but it's the reference implementation from discourse: https://github.com/discourse/discourse/blob/master/lib/single_sign_on.rb

I opted out of making any changes (except for the necessary ones for our environment) to make any troubleshooting we might need to do easier. I dropped a comment here noting this: https://github.com/OperationCode/operationcode_backend/blob/3b98c8371a351a2b20f3727ebc9bfbcc4b7e6001/app/lib/discourse/single_sign_on.rb#L4

**Pull request merged**